compiler/src/main/scala/org/scalajs/nscplugin/JSGlobalAddons.scala

```diff
@@ -131,6 +132,122 @@ trait JSGlobalAddons extends JSDefinitions
       }
     }
 
+    sealed abstract class JSCallingConvention {
+      def displayName: String
+    }
+
+    object JSCallingConvention {
+      case object Call extends JSCallingConvention {
+        def displayName: String = "function application"
+      }
+
+      case object BracketAccess extends JSCallingConvention {
+        def displayName: String = "bracket access"
+      }
+
+      case object BracketCall extends JSCallingConvention {
+        def displayName: String = "bracket call"
+      }
+
+      case class Method(name: JSName) extends JSCallingConvention {
+        def displayName: String = "method '" + name.displayName + "'"
+      }
+
+      case class Property(name: JSName) extends JSCallingConvention {
+        def displayName: String = "property '" + name.displayName + "'"
+      }
+
+      case class UnaryOp(code: js.JSUnaryOp.Code) extends JSCallingConvention {
+        def displayName: String = "unary operator"
+      }
+
+      case class BinaryOp(code: js.JSBinaryOp.Code) extends JSCallingConvention {
+        def displayName: String = "binary operator"
+      }
+
+      def of(sym: Symbol): JSCallingConvention = {
+        require(sym.isTerm)
+
+        if (isJSBracketAccess(sym)) {
+          BracketAccess
+        } else if (isJSBracketCall(sym)) {
+          BracketCall
+        } else {
+          val jsNameAnnot = sym.getAnnotation(JSNameAnnotation)
+          lazy val pc = sym.paramss.map(_.size).sum
+
+          sym.name match {
+            case nme.apply if jsNameAnnot.isEmpty => Call
+
+            case JSUnaryOpMethodName(code) if pc == 0 && jsNameAnnot.isEmpty =>
```

Comment:
Only doing this conditional on the parameter count is the current behavior. I wonder if we should keep it.